src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,74 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int AMMOUNT_INDEX = 1;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] data = readFromFile(fromFileName);
+        int totalResult = calculateTotalSupply(data) - calculateTotalBuy(data);
+        String reportStatistic =
+                "supply" + "," + calculateTotalSupply(data) + System.lineSeparator()
+                + "buy" + "," + calculateTotalBuy(data) + System.lineSeparator()
+                + "result" + "," + totalResult;
+        writeToFile(reportStatistic, toFileName);
+    }
 
+    private int calculateTotalSupply(String[] data) {
+        int totalSupply = 0;
+        for (String line : data) {
+            String[] splittedLine = line.split(",");
+            if (splittedLine[OPERATION_TYPE_INDEX].equals("supply")) {
+                totalSupply += Integer.parseInt(splittedLine[AMMOUNT_INDEX]);
+            }
+        }
+        return totalSupply;
+    }
+
+    private int calculateTotalBuy(String[] data) {
+        int totalBuy = 0;
+        for (String line : data) {
+            String[] splittedLine = line.split(",");
+            if (splittedLine[OPERATION_TYPE_INDEX].equals("buy")) {
+                totalBuy += Integer.parseInt(splittedLine[AMMOUNT_INDEX]);
+            }
+        }
+        return totalBuy;
+    }
```

Comment:
we neet to go through all data twice here, could we make it only with one loop? (for this we neet co combine these 2 methods in one)

Reply:
BuyAndSupply in one method